### PR TITLE
#6115 Versioning, build, dacpac generation

### PR DESCRIPTION
**Pull request by adoprog**
- Versioning
 - Tagging
 - Build & Push (Azure DevOps)
 - Sitecore Courier / DacPac generation

image

### CODE REVIEW

**Review by jflheureux — CHANGES_REQUESTED** [review]
Great work. Found a couple of issues and multiple improvement ideas.

File: init-ci.ps1 [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    HelpMessage = "Process Isolation to use when building images")]
+  [string]$IsolationMode = "hyperv"
+  ,
+  [Parameter(
+    HelpMessage = "Windows image version")]
+  [string]$WindowsVersion = "ltsc2019"
+  ,
+  [Parameter(
+    HelpMessage = "Sitecore version")]
+  [string]$SitecoreVersion = "10.1.0"
```
[the remark this answers is not in the document]
Comment (adoprog) [reply init-ci.ps1:24]: Updated .env to use 10.1.0

File: docker-compose.override.yml [hunk 2]
```diff
@@ -74,14 +74,14 @@ services:
...
-    isolation: ${ISOLATION}
-    image: ${REGISTRY}${COMPOSE_PROJECT_NAME}-xm1-mssql-init:${VERSION:-latest}
+    image: ${REGISTRY}demo/${COMPOSE_PROJECT_NAME}-mssql-init:${SITECORE_VERSION}-${WINDOWSSERVERCORE_VERSION}-${DEMO_VERSION:-latest}
     build:
       context: ./docker/build/mssql-init
       args:
-        PARENT_IMAGE: scr.sitecore.com/sxp/sitecore-xm1-mssql-init:10.1-ltsc2019
-        HEADLESS_SERVICES_IMAGE: ${HEADLESS_SERVICES_IMAGE}
-        SPE_ASSETS_IMAGE: ${SPE_ASSETS_IMAGE}
+        PARENT_IMAGE: scr.sitecore.com/sxp/sitecore-xm1-mssql-init:10.1-${WINDOWSSERVERCORE_VERSION}
```
[the remark this answers is not in the document]
Comment (jflheureux) [reply docker-compose.override.yml]: `scr.sitecore.com/` should be replaced by `${SITECORE_DOCKER_REGISTRY}` too

File: init-ci.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (jflheureux) [reply init-ci.ps1:24]: I think this has not been committed.

File: docker-compose.override.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (adoprog) [reply docker-compose.override.yml]: Done.

File: init-ci.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (adoprog) [reply init-ci.ps1:24]: Right, thanks. pushed it now.

**Review by jflheureux — APPROVED** [review]

**Pull request merged**